### DISCUSSION TITLE
How does git fetch functionality work?

### DISCUSSION THREAD

**Discussion by rbren** [starship/starship#1640] [Q&A] [answered]
Does starship periodically run `git fetch`? Or is it up to the user to run `git fetch`?

**Comment by matchai** [comment, marked as answer]
Starship doesn't run `git fetch` as that would delay the rendering of the prompt until a response is received.
We only show the git status of what is already locally available.

**Comment by rbren** [comment]
Thanks! #301 and #1551 seem relevant

**Comment by rbren** [comment]
FWIW, I've hacked in a periodic, background `git fetch` that runs without blocking the prompt

```bash
function maybe_git_fetch () {
  if ! [ -d ".git" ]; then
    return
  fi
  last_fetch=$(stat -c %Y .git/FETCH_HEAD)
  time_now=$(date +%s)
  timeout=60
  if [[ $((time_now - timeout)) -gt $((last_fetch)) ]]; then
    (git fetch 2> /dev/null &)
  fi
}
export PROMPT_COMMAND="maybe_git_fetch"
```

**Reply by nwaywood** [reply]
For those using zsh, I just yoinked this (you don't need to be using oh-my-zsh for it to work 🎉 ) https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/git-auto-fetch

**Comment by Micha-Btz** [comment]
Hey,
I didn't understand the way rbren did it, so I tried
[custom.git]
when = "git status"
command = "git fetch"
shell = ["bash"]

but it seems to do nothing. I have only some small repos so the time fetch takes doesn't care me.
For me I don't need a periodic test, only if I change to the dir first time a git fetch would be enough for me.
Can you explain, why me command is not  working?

TIA Micha
